lib/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -84,4 +91,10 @@ public UserIdentityService(
 
         return userIssuedDebugCredentials;
     }
+
+    private boolean isNotPopulatedJsonArray(Object input) {
```

Comment:
This was hard to get my head around.

Are we testing that the array is empty, but exists within the vcClaim object?

So an empty `{}`

Would `vcClaim.get(VC_EVIDENCE).length() == 0` work instead? Rather than testing the instances etc.

Reply:
Yeah, this is a bit of a mind bender. And the name of it isn't super helpful!

It returns true if:

- The input in null
- The input isn't a JSONArray
- The JSONArray is empty
- The first element in the JSONArray isn't a JSONObject.

It's basically doing a safety check to make sure that value for the evidence attribute in the verifiable credential exists and is an object, before we try and access it.
I hate working with JSON in Java.

Reply:
> I hate working with JSON in Java.

I agree - this looks good then!